lib/spack/env/cc

```diff
@@ -142,12 +167,27 @@ append() {
 # to the list stored in LISTNAME1.
 # If PREFIX is provided, prepend it to each element.
 extend() {
+    if eval "[ -z \"\$$2\" ]"; then
+        return
+    fi
+    # Figure out the appropriate IFS for the list we're writing.
+    setsep "$1"
+    osep="$sep"
     # Figure out the appropriate IFS for the list we're reading.
     setsep "$2"
     if [ "$sep" != " " ]; then
         IFS="$sep"
     fi
-    eval "for elt in \${$2}; do append $1 \"$3\${elt}\"; done"
+    # if separators match, and no prefix, just append
+    if [ -z "$3" -a "$sep" = "$osep" ]; then
+        appendsv "$1" "$2" "$osep"
+        return
+    fi
```

Comment:
Just to clarify: I'm reading this as for performance only, otherwise the for-loop would take care of it?

Reply:
Correct. The append is significantly cheaper, and when I tested it it's also the most common case.